src/System.Data.SqlClient/tests/FunctionalTests/SqlConnectionTest.cs

```diff
@@ -30,6 +30,29 @@ public void ConnectionTest()
             Assert.Null(e);
         }
 
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]  // Integ auth on Test server is supported on Windows right now
+        public void IntegratedAuthConnectionTest()
+        {
+            Exception e = null;
+
+            using (TestTdsServer server = TestTdsServer.StartTestServer())
+            {
+                try
+                {
+                    SqlConnectionStringBuilder builder = new SqlConnectionStringBuilder(server.ConnectionString);
+                    builder.IntegratedSecurity = true;
+                    SqlConnection connection = new SqlConnection(builder.ConnectionString);
+                    connection.Open();
+                }
+                catch (Exception ce)
+                {
+                    e = ce;
+                }
+            }
+            Assert.Null(e);
```

Comment:
I had added this kind of behavior to make it explicit that the test expects that no exception is thrown. A test without an Assert seemed incomplete. The question I had was "How do you communicate from the test constructs what is being tested?" And this code is what I came up with.

This is not the best way a test for Integrated Auth can be written as it doesn't verify the integrated auth specific pieces. What I really want to test is that the SqlClient's generated auth tokens can be validated by the server and the number of round trips to the server from the client in case of a successful token validation. I cannot access this data from SqlClient APIs.

@stephentoub After you made this comment, I read a bit more about what others have to say about a test without asserts and there are philosophical differences in the way test code is written and an Assert to check nothing went wrong was for readability. 
This is the reason the `AssertNotThrows` was removed from XUnit. (Xunit issue 188)

Ideally there should be no test which simply runs code and says "Nothing wrong happened and everything looks good". Since we don't have a way of testing these from the internals of the SqlClient I have to resort to a 'nothing wrong happened' paradigm. 

I am leaning more towards, remove the Assert, run the code and don't use assert in case no exception was expected. I will see what @corivera and @geleems have to say about it as well before I make the change.